feat: add filtering by container label for Docker integration

gntouts opened the pull request:
This commit introduces the ability to filter Docker containers by their labels, instead of container names, in the Docker integration. This functionality allows dynamic container name resolution, which is particularly useful in environments where container names may change frequently or are not predefined.

- Added container_label field to the dockerConfig type
- Updated DockerIntegration.isAvailable()  to deduce the container name from `container_label` if the container name is not provided
- Updated configuration example and documentation to include the new feature

This is my first time writing TypeScript, so I'd appreciate any feedback or suggestions on improving the code, especially regarding style or best practices.

Fixes #193

tale reviewed (approved):
Looks good overall, one thought I had is if we should suggest container labels to be the default method going forward. It's a better discovery path, but we leave the container name option available and commented out by default.

gntouts commented:
Thanks for the review! I agree — using container labels as the default method sounds like a good idea for improved flexibility. I updated the configuration example to use `container_label` as the default and leave the `container_name` option commented out for reference. I also updated the docs.

tale commented:
Looks good, though one last change before merging in. Can you update the docker compose example in the docs to label the headscale container to match the default config? Once that's done I'll merge this into the pre-release tag so it's testable.

tale commented:
Thank you.
